src/windows/cpu.rs

```diff
@@ -503,6 +507,109 @@ pub(crate) fn get_frequencies(nb_cpus: usize) -> Vec<u64> {
     vec![0; nb_cpus]
 }
 
+pub(crate) fn get_realtime_freq() -> f64{
+    if let Ok(avg) = realtime_freq().read() {
+        if let Some(avg) = *avg {
+            return avg;
+        }
+    }
+    0.0
+}
+
+fn realtime_freq() -> &'static RwLock<Option<f64>> {
+    static REALTIME_FREQUENCY: OnceLock<RwLock<Option<f64>>> = OnceLock::new();
+    REALTIME_FREQUENCY.get_or_init(|| unsafe { init_realtime_freq() })
+}
+
+unsafe extern "system" fn update_realtime_freq_callback(counter: *mut c_void, _: BOOLEAN) {
+    let val = parse_cpu_realtime_freq_counter(counter);
+    if let Ok(mut freq) = realtime_freq().write() {
+        if let Some(freq) = freq.deref_mut() {
+            *freq = val;
+        }
+    }
+}
+
+unsafe fn init_realtime_freq() -> RwLock<Option<f64>> {
+    let mut query = 0;
+
+    if PdhOpenQueryA(PCSTR::null(), 0, &mut query) != ERROR_SUCCESS.0 {
+        sysinfo_debug!("init_cpu_realtime_frequency: PdhOpenQueryA failed");
+        return RwLock::new(None);
+    }
+
+    let mut cpu_freq_counter = 0;
+    if PdhAddEnglishCounterA(
+        query,
+        s!("\\Processor Information(_Total)\\% Processor Performance"),
+        0,
+        &mut cpu_freq_counter,
+    ) != ERROR_SUCCESS.0
+    {
+        PdhCloseQuery(query);
+        sysinfo_debug!("init_cpu_realtime_frequency: failed to get CPU performance");
+        return RwLock::new(None);
+    }
+
+    // We want to be able to get the value directly by calling the function,
+    // so we collect it twice in this init function
+    PdhCollectQueryData(query);
+    thread::sleep(Duration::from_millis(100));
```

Comment:
This is pretty bad. Can't we just do like CPU usage and tell in the doc that the first reading will be likely wrong?